hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.tsx

```diff
@@ -0,0 +1,356 @@
+            ))}
+        </>
+    );
+};
+
+interface MainSectionProps {
+    name: string;
+    quantityAvailable: number;
+    quantityRemaining: number;
+    categories: string[];
+    picture: string;
+}
+const MainSection = ({
+    name,
+    quantityAvailable,
+    quantityRemaining,
+    categories,
+    picture,
+}: MainSectionProps) => {
+    const availability =
+        quantityRemaining === 0 ? (
+            <Typography color="secondary">OUT OF STOCK</Typography>
+        ) : (
+            <Typography className={styles.quantityAvailable}>
+                {quantityRemaining} OF {quantityAvailable} IN STOCK
+            </Typography>
+        );
+
+    return (
+        <div className={styles.mainSection}>
+            <div>
+                <Typography variant="h6">{name}</Typography>
+                {availability}
+                <Typography variant="body2" className={styles.heading}>
+                    Category
+                </Typography>
+                <div>
+                    {categories?.length > 0 &&
+                        categories.map((category, i) => (
+                            <Chip
+                                label={category}
+                                size="small"
+                                className={styles.categoryItem}
+                                key={i}
+                            />
+                        ))}
+                </div>
+            </div>
+            <img src={picture} alt={name} />
+        </div>
+    );
+};
+
+export const ProductOverview = ({
+    showAddToCartButton,
+}: {
+    showAddToCartButton: boolean;
+}) => {
+    let categoryNames: string[] = [];
+    let maxPerTeam: number | null = null;
+    let constraints: string[] = [];
+
+    const dispatch = useDispatch();
+    const closeProductOverview = () => dispatch(removeProductOverviewItem());
+
+    const isProductOverviewVisible: boolean = useSelector(
+        isProductOverviewVisibleSelector
+    );
+    const hardware = useSelector(hardwareBeingViewedSelector);
+    const categories = useSelector((state: RootState) =>
+        selectCategoriesByIds(state, hardware?.categories || [])
+    );
+    if (categories.length > 0) {
+        categoryNames = categories.map(
+            (category) => category?.name ?? `Category ${category?.id}`
+        );
+        categories.filter((category) => !!category).map((category) => category?.name);
+
+        constraints = hardware?.max_per_team
+            ? [`Max ${hardware.max_per_team} of this item`]
+            : [];
+        maxPerTeam = hardware?.max_per_team ?? Infinity;
+        for (const category of categories) {
+            if (category?.max_per_team) {
+                constraints.push(
+                    `Max ${category.max_per_team} of items under category ${category.name}`
+                );
+                maxPerTeam = Math.min(category.max_per_team, maxPerTeam);
+            }
+        }
```

Comment:
Change the declaration of `maxPerTeam` to `let maxPerTeam: number | undefined;`. Then:

```suggestion
        maxPerTeam = hardware?.max_per_team;
        for (const category of categories) {
            if (category?.max_per_team) {
                constraints.push(
                    `Max ${category.max_per_team} of items under category ${category.name}`
                );
                maxPerTeam =
                    maxPerTeam === undefined
                        ? category.max_per_team
                        : Math.min(category.max_per_team, maxPerTeam);
            }
        }
```

The use of Infinity otherwise concerns me

Reply:
You can keep it as `null` if you want instead of `undefined`, just initialize it as such and ignore the rest of my comments on the matter.